src/Modules/Legacy/Inverse/SolveInverseProblemWithTikhonovSVD.cc

```diff
@@ -141,6 +141,53 @@ void SolveInverseProblemWithTikhonovSVD::execute()
 		sendOutputFromAlgorithm(InverseSolution,output);
 		sendOutputFromAlgorithm(RegularizationParameter,output);
 		sendOutputFromAlgorithm(RegInverse,output);
+    auto lambda=output.get<DenseMatrix>(TikhonovAlgoAbstractBase::RegularizationParameter);
+    auto lambda_array=output.get<DenseMatrix>(TikhonovAlgoAbstractBase::LambdaArray);
+    auto lambda_index =output.get<DenseMatrix>(TikhonovAlgoAbstractBase::Lambda_Index);
+    
+    auto regularization_method  = state->getValue(Parameters::RegularizationMethod).toString();
+    
+    if (regularization_method== "lcurve") update_lcurve_gui(lambda,lambda_array,lambda_index);
 
 	}
 }
+
+void SolveInverseProblemWithTikhonovSVD::update_lcurve_gui(const  DenseMatrixHandle& lambda, const DenseMatrixHandle& input, const  DenseMatrixHandle& lambda_index)
+{
+  double *l_data = lambda->data();
+  double lamb =l_data[0];
+  auto state = get_state();
+  state->setValue(Parameters::LambdaCorner, lamb);
+  double *li_data = lambda->data();
+  int lam_ind = static_cast<int>(li_data[0]);
+  
+  size_t nLambda = input->rows();
+  size_t nr = input->cols();
+  std::vector<double> eta(nLambda);
+  std::vector<double> rho(nLambda);
+  //std::vector<double> lambdaarray(nLambda);
+  
+  double *in_data = input->data();
+  for (int k=0; k<nLambda; k++)
+  {
+    //lambdaarray=in_data[k*nr];
+    rho[k]=in_data[k*nr+1];
```

Comment:
There are slicker ways to do this, with Eigen slices (or just pull out a column and store in a vector)